httomo/wrappers_class.py

```diff
@@ -0,0 +1,237 @@
+from typing import Dict
+import numpy as np
+import inspect
+from mpi4py.MPI import Comm
+from inspect import signature
+
+gpu_enabled = False
+try:
+    import cupy as xp
+
+    try:
+        xp.cuda.Device(0).compute_capability
+        gpu_enabled = True  # CuPy is installed and GPU is available
+    except xp.cuda.runtime.CUDARuntimeError:
+        import numpy as xp
+
+        print("CuPy is installed but GPU device inaccessible")
```

Comment:
I am guessing you are using this attribute to check if `Device(0)` is accessible -- makes sense.

Reply:
The point is, you wanna try making one call to the CUDA runtime library to make sure it is possible, and there are administrative settings etc that can also block you from using a GPU. All of these that I'm aware of would limit the device count visible to the process, so checking device count is likely enough. 

For example, an admin might block the process from using any GPU, in which case the `getDeviceCount()` call would just return 0.

Of course there is the case that the GPU is somehow busy or faulty - if that's the case, neither method will work and it would simply crash further down the line. But that's rare enough and I wouldn't try to guard against that. (We don't check whether CPU RAM is working or cores are working before using them either)

Reply:
yes, I do aim to get the CUDARuntimeError if there is no device 0 available. The zeroth device should be present as a starting point requirement to use a GPU. The number of devices is a minor importance at the stage I think...